docs/rfc/rfc-103-incoming-txs-when-catching-up.md

```diff
@@ -0,0 +1,177 @@
+chain. When a node is performing BlockSync or StateSync, it is in *catch up*
+mode. Once synchronization finishes, the node is ready to fully participate in
+consensus, so it switches to *consensus* mode, which should never leave unless
+the node crashes.
+
+### Transaction validation with CheckTx
+
+The `CheckTx` [method](#check-tx) acts as a gatekeeper to the mempool, which
+only accepts valid transactions. The validity of a transaction is defined
+deterministically by the application, but it may depend on the application
+state, which may change at every height.
+
+`CheckTx` is called:
+- from the [RPC endpoints](#rpc) `broadcast_tx_async`, `broadcast_tx_sync`, and
+  `broadcast_tx_commit`, and
+- when the node [receives](#receive) a `Txs` message from a peer: for each
+  transaction in `Txs` there will be a call to `CheckTx`.
+
+`CheckTx` puts the transaction in the cache, sends a `RequestCheckTx` ABCI
+message to the application, and sets a callback to process a `ResponseCheckTx`.
+If the application deems the transaction as valid, `CheckTx` stores the
+transaction in the mempool and notifies the consensus reactor that there are
+transactions available to be [put in a new block](#reap). Otherwise it does not
+add the transaction to the mempool and removes it from the cache.
+
+### Mempool cache
+
+The mempool implements a cache of received transactions that it uses to discard
+already-seen transactions, thus avoiding processing duplicates.
+
+Transactions remain in the cache until the application replies in
+`ResponseCheckTx` that a transaction is invalid. This can occur the first time a
+transaction is received and processed by `CheckTx`, or when all the transactions
+in the mempool need to be rechecked after updating the mempool. Another instance
+is when the consensus reactor [updates](#update) the mempool after finalizing a
+block: if there was an error while executing a transaction, as indicated by the
+application in `ResponseFinalizeBlock`, then the transaction is removed from the
+cache.
+
+All these cases are for removing invalid transactions, so the cache also
+requires that `keep-invalid-txs-in-cache` is set to false. The configuration
+`keep-invalid-txs-in-cache` is for ignoring invalid, already-seen transactions.
+This is useful for applications that consider rejected transactions will never
+be accepted again.
+
+If a received transaction is valid but the mempool is full, the transaction is
+not kept in the cache. This is to give the transaction a chance to be
+re-validated at a later moment with free space in the mempool.
+
+## Discussion
+
+### Inbound messages
+
+A node that is not in consensus mode, should accept or reject incoming
+transactions? Currently the mempool accepts and process all incoming
+transactions. It only implements the cache to discard already-seen transactions.
+
+There are not many reasons at this stage to have an operational mempool reactor
```

Comment:
Rejecting inbound requests to `broadcast_tx_*` before the node is in consensus mode makes sense! 

It sounds like the node cannot offer any guarantees (even participation in gossip to propagate transactions is not guaranteed as far as I understand) that can be offered before consensus mode is active.

I think rejecting requests to `broadcast_tx_*` until some guarantees can be provided for those requests -- once consensus is active and node is not `catching_up = true` -- would improve the experience of users of this RPC because the semantics will be much clearer.

Reply:
It also makes sense to double-check with some users on this. For that:
- I will mention it during the community call next week (may 11) and ask feedback.
- I will draw attention of some users (relayer operators, validators, integrators) here.